pkg/eval/compile_value_test.go

```diff
@@ -73,8 +73,9 @@ func TestMapLiteral(t *testing.T) {
 func TestStringLiteral(t *testing.T) {
 	Test(t,
 		That(`put 'such \"''literal'`).Puts(`such \"'literal`),
-		That(`put "much \n\033[31;1m$cool\033[m"`).
-			Puts("much \n\033[31;1m$cool\033[m"),
+		That(`put "much \n\033[31;1m$cool\033[m"`).Puts("much \n\033[31;1m$cool\033[m"),
+		That(`put "x\302\222\302\214y"`).Puts("x\u0092\u008cy"),
+		That(`put "x\u0092\u008cy"`).Puts("x\u0092\u008cy"),
```

Comment:
I still don't get why these two extra tests belong here; the interpretation of escape sequences is done in the parser, not in this package.

The existing test on line 76 does test for escape sequences, but that's intended to be a sanity check of whether double-quoted strings work as expected, and is not intended to be an in-depth test of escape sequences in general.

Reply:
Yes, and no. I agree the second test isn't meaningful but believe the first test, of a sequence of valid UTF-8 bytes, is useful. I thought I had verified that both tests failed without the other changes in this PR but that is not true. Nonetheless, I think the first test is useful to help protect against future regressions.

Reply:
The code these test cases exercise are already covered by unit tests in the `parse` package. Not sure which regressions you are intending to guard against.

I am not against having more test cases, but test cases need to have clear intentions on which part of the code they are supposed to guard.